mods/boats/init.lua

```diff
@@ -180,15 +171,13 @@ function boat.on_step(self, dtime)
 		self.object:set_pos(self.object:get_pos())
 		return
 	end
-	local s = get_sign(self.v)
-	self.v = self.v - dtime * 0.6 * s
-	if s ~= get_sign(self.v) then
-		self.object:set_velocity({x = 0, y = 0, z = 0})
+	-- We need to preserve velocity sign to properly apply drag force while moving backward
```

Comment:
This line is just over 90 columns (tab size 4), please wrap.

Reply:
Done